AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/Model+Compare.swift

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+// swiftlint:disable cyclomatic_complexity
+// swiftlint:disable syntactic_sugar
+extension ModelSchema {
+
+    /// Compare two `Model` based on a given `ModelSchema`
+    /// Returns true if equal, false otherwise
+    public static func isEqual(_ model1: Model,
+                               _ model2: Model,
+                               _ modelSchema: ModelSchema) -> Bool {
+        for (fieldName, modelField) in modelSchema.fields {
+            // skip read only fields
```

Comment:
good place to add comment to why we skip read-only fields; like `updatedAt` is created/updated by the service so for the purpose of this equality check, we ignore it. what are the read-only fields other than updatedAt? does it hold true for those other fields as well?